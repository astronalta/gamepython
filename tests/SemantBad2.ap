# Filler
# Filler

Sub1 < Object {

    foo Sub1::Test
}

import Sub2

test() Bool {
# This function should override Sub2::test() which rets Int
    ret 
}

main() Int {
    ret test()
}

Test < Object {
# This class should override Sub2::Test, which has type bob
    bob() {
    }
}

main2() Int {
    Test().bob()
    ret 0
}

native_function1() native {
    # OK
}

native_function2() native Int {
    ret 1 # Not OK
}

native_function3() native # OK
  
Obj1 < Object {
    bob Int = bill
    bill Int = bob
    obj2 Obj2
    one Int = obj2.two
}

Obj2 < Object {
    obj1 Obj1
    two Int = obj1.one
    test = bob
    bob = test
}

