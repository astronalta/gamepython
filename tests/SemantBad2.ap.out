SemantBad2.ap:4:1: Type name 'Sub1' conflicts with an existing module name
SemantBad2.ap:11:1: Function 'test' must return a value
SemantBad2.ap:17:5: Return must conform to type 'Int'
SemantBad2.ap:35:1: Native function 'native_function2' has a body
SemantBad2.ap:43:16: Illegal circular reference
SemantBad2.ap:52:15: Illegal circular reference
SemantBad2.ap:47:14: Illegal circular reference
SemantBad2.ap:54:11: Illegal circular reference
